debian/jitsi-meet-turnserver.postinst

```diff
@@ -0,0 +1,108 @@
+#!/bin/bash
+# postinst script for jitsi-meet-turnserver
+#
+# see: dh_installdeb(1)
+
+set -e
+
+# summary of how this script can be called:
+#        * <postinst> `configure' <most-recently-configured-version>
+#        * <old-postinst> `abort-upgrade' <new version>
+#        * <conflictor's-postinst> `abort-remove' `in-favour' <package>
+#          <new-version>
+#        * <postinst> `abort-remove'
+#        * <deconfigured's-postinst> `abort-deconfigure' `in-favour'
+#          <failed-install-package> <version> `removing'
+#          <conflicting-package> <version>
+# for details, see http://www.debian.org/doc/debian-policy/ or
+# the debian-policy package
+
+case "$1" in
+    configure)
+        # loading debconf
+        . /usr/share/debconf/confmodule
+
+        # try to get host from jitsi-videobridge
+        db_get jitsi-videobridge/jvb-hostname
+        if [ -z "$RET" ] ; then
+            # server hostname
+            db_set jitsi-videobridge/jvb-hostname "localhost"
+            db_input critical jitsi-videobridge/jvb-hostname || true
+            db_go
+        fi
+        JVB_HOSTNAME="$RET"
+
+        TURN_CONFIG="/etc/turnserver.conf"
+
+        # detect dpkg-reconfigure, just delete old links
+        db_get jitsi-meet-turnserver/jvb-hostname
+        JVB_HOSTNAME_OLD=$RET
+        if [ -n "$RET" ] && [ ! "$JVB_HOSTNAME_OLD" = "$JVB_HOSTNAME" ] ; then
+            rm -f $TURN_CONFIG
+        fi
+
+        # stores the hostname so we will reuse it later, like in purge
+        db_set jitsi-meet-turnserver/jvb-hostname "$JVB_HOSTNAME"
+
+        # try to get turnserver password
+        db_get jitsi-meet-prosody/turn-secret
+        if [ -z "$RET" ] ; then
+            db_input critical jitsi-meet-prosody/turn-secret || true
+            db_go
+        fi
+        TURN_SECRET="$RET"
+
+        if [ ! -f $TURN_CONFIG ] ; then
+            PUBLIC_IP=$(dig +short myip.opendns.com @resolver1.opendns.com)
+            cp /usr/share/jitsi-meet-turnserver/turnserver.conf $TURN_CONFIG
+            sed -i "s/jitsi-meet.example.com/$JVB_HOSTNAME/g" $TURN_CONFIG
+            sed -i "s/__turnSecret__/$TURN_SECRET/g" $TURN_CONFIG
+            sed -i "s/__external_ip_address__/$JVB_HOSTNAME/g" $TURN_CONFIG
+
+            # SSL for nginx
+            db_get jitsi-meet/cert-choice
+            CERT_CHOICE="$RET"
+
+            if [ "$CERT_CHOICE" = "I want to use my own certificate" ] ; then
```

Comment:
we don't support let's encrypt for this yet, right? also, how do browsers behave with self-signed certs on TURN servers?

Reply:
It uses the same certs as nginx and same domain so we re-use the certs. I think it doesn't work with self-signed.

Reply:
Ack.